apps/ejabberd/src/mongoose_admin.erl

```diff
@@ -0,0 +1,264 @@
+            {args, [{jid, binary}]},
+            {result, {msg, binary}}
+        ],
+%%        [
+%%            {name, getuserresources},
+%%            {category, session},
+%%            {desc, "Get user resources"},
+%%            {module, ?MODULE},
+%%            {function, get_user_resources},
+%%            {action, read},
+%%            {args, [{jid, binary}]},
+%%            {result, []}
+%%        ],
+        [
+            {name, listsessions},
+            {category, session},
+            {desc, "Get session list"},
+            {module, ?MODULE},
+            {function, listsessions},
+            {action, read},
+            {args, [{host, binary}]},
+            {result, []}
+        ],
+        [
+            {name, kickuser},
+            {category, session},
+            {desc, "Terminate user connection"},
+            {module, ?MODULE},
+            {function, kick_this_session},
+            {action, delete},
+            {args, [{jid, binary}]},
+            {result, {msg, binary}}
+        ],
+        [
+            {name, sendmessage},
+            {category, message},
+            {desc, "Send chat message from to"},
+            {module, ?MODULE},
+            {function, send_message},
+            {action, create},
+            {security_policy, [user]},
+            {args, [{caller, binary}, {to, binary}, {msg, binary}]},
+            {result, ok}
+        ],
+        [
+            {name, getmessages},
+            {category, message},
+            {desc, "Get recent messages"},
+            {module, ?MODULE},
+            {function, get_recent_messages},
+            {action, read},
+            {security_policy, [user]},
+            {args, [{caller, binary}, {other, binary}, {limit, integer}]},
+            {result, []}
+        ],
+        [
+            {name, changepassword},
+            {category, user},
+            {desc, "Change user password"},
+            {module, ?MODULE},
+            {function, change_user_password},
+            {action, update},
+            {security_policy, [user]},
+            {identifiers, [caller]},
+            {args, [{caller, binary}, {newpass, binary}]},
+            {result, ok}
+        ]
+    ].
+
+%%get_user_resources(Jid) ->
+%%    J = jid:from_binary(Jid),
+%%    ejabberd_sm:get_user_resources(J#jid.user, J#jid.server).
+
+
+kick_this_session(Jid) ->
+    J = jid:from_binary(Jid),
+    ejabberd_sm:route(
+        jid:make(<<"">>, <<"">>, <<"">>),
+        J,
+        {broadcast, {exit, <<"kicked">>}}),
+    <<"kicked">>.
+
+mk(E) ->
+    Ss = element(1, E),
+    {U, S, R} = Ss,
+    <<U/binary, "@", S/binary, "/", R/binary>>.
+
+listsessions(Host) ->
+    Lst = ejabberd_sm:get_vh_session_list(Host),
+    lists:map(fun mk/1, Lst).
+
+
+registered_users(Host) ->
+    Users = ejabberd_auth:get_vh_registered_users(Host),
+    SUsers = lists:sort(Users),
+    lists:map(fun({U, S}) -> <<U/binary, "@", S/binary>> end, SUsers).
+
+
+register(User, Host, Password) ->
```

Comment:
In order to be consistent with the code base I suggest changing the params order so `Host` is before `User` as in other places.